diff --git a/bin/ROSE_main.py b/bin/ROSE_main.py
index 761be5f..9416900 100755
--- a/bin/ROSE_main.py
+++ b/bin/ROSE_main.py
@@ -307,7 +307,7 @@ def main():
     inputName = inputGFFFile.split('/')[-1].split('.')[0]
 
     #GETTING THE CORRECT ANNOT FILE
-    cwd = os.getcwd()
+    cwd = os.path.abspath(os.path.dirname(__file__) + '/../ROSE/annotation')
     genomeDict = {
         'HG18':'%s/annotation/hg18_refseq.ucsc' % (cwd),
         'MM9': '%s/annotation/mm9_refseq.ucsc' % (cwd),
